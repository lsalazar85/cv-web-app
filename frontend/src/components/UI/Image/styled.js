import styled from 'styled-components'

const ImageContainer = styled.div`
    width: ${props => props.width};
    height: auto;
`

const ImageElement = styled.img`
  width: 100%;
  height: auto;
  border-radius:${props => props.circular && '50%'};
`

export { ImageContainer, ImageElement }