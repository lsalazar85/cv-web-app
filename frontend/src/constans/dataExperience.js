export const DataExperience = [
  {
    id: 1,
    title: 'Zx Ventures',
    years: 'May 2021 - Present',
    occupation: 'Front End Software Engineer',
    linkSite: 'https://www.zx-ventures.com/',
    content: [
      {key: 1, text: 'Implemented new functions for Siempre en Casa app in Argentina, improving the process of configuring plans for the purchase of beverages, shipping status, and selection of new products, and the new UI helped reach 40.000 active users.'},
    ],
    techStack: [
      {key: 1, label: 'Typescript'},
      {key: 2, label: 'Next.js'},
      {key: 3, label: 'Styled Components'},
      {key: 4, label: 'Jest & Enzyme'},
      {key: 5, label: 'Git & Github'},
      {key: 6, label: 'Jenkins'},
      {key: 7, label: 'Codacy'},
    ],
  },
  {
    id: 2,
    title: 'Mercado Libre',
    years: 'January 2020 - May 2021',
    occupation: 'Software Engineer',
    linkSite: 'https://www.mercadolibre.com',
    content: [
      {key: 1, text: 'Participated in the team for the shipping area of Mercado Env√≠os (Main Shipping Management).'},
      {key: 2, text: 'Performed code reviews on the frontend team by promoting code quality.'},
      {key: 3, text: 'Contributed to building 2 web operational apps for the main warehouses:'},
      {key: 4, text: 'Custom Offsets: Improved performance by 99% for reconfiguration of delayed shipments.'},
      {key: 5, text: 'Transit Time: Programmed multiple routes and origins for different types of shipments, reduced route bugs by 90% from the main warehouses in Mexico and Colombia.'},
    ],
    techStack: [
      {key: 1, label: 'JavaScript'},
      {key: 2, label: 'React.js'},
      {key: 3, label: 'Node.js & Express.js (BFF)'},
      {key: 4, label: 'Jest & Enzyme' },
      {key: 5, label: 'React Testing Library'},
      {key: 6, label: 'Styled Components'},
      {key: 7, label: 'Git & Github'},
      {key: 8, label: 'Docker for local setup'},
      {key: 9, label: 'Jenkins for CI/CD'},
    ]
  },
  {
    id: 3,
    title: 'Rappi',
    years: 'August 2019 - January 2020',
    occupation: 'Front End Developer',
    linkSite: 'https://www.rappi.com',
    content: [
      {key:1, text: 'Developed an internal app to configure different food dishes weekly, with more accessible prices for the general public. This implementation was exclusive to Colombia. In the first month, 100,000 orders were successfully configured.'},
    ],
    techStack: [
      {key: 1, label: 'JavaScript'},
      {key: 2, label: 'React.js'},
      {key: 3, label: 'CSS Modules'},
      {key: 4, label: 'Mobile First'},
      {key: 5, label: 'Git & Github'},
      {key: 6, label: 'Jenkins for CD/CI.'},
    ],
  },
  {
    id: 4,
    title: 'Global Hitts Cono Sur',
    years: 'February 2019 - July 2020',
    occupation: 'Front End Developer',
    linkSite: 'https://globalhitssconosur.com/',
    content: [
      {key: 1, text: 'Developed and implemented a new Claro Video  (Like Netflix app) implementation in Brazil.'},
      {key: 2, text: 'Improved the user experience in the application to watch movies and sports events from any device.'}
    ],
    techStack: [
      {key: 1, label: 'JavaScript'},
      {key: 2, label: 'React.js'},
      {key: 3, label: 'Redux'},
      {key: 4, label: 'Sass'},
      {key: 5, label: 'Git & Github'},
      {key: 6, label: 'Material-UI'},
    ],
  },
  {
    id: 5,
    title: 'BlueAlba',
    years: 'April 2018 - February 2019',
    occupation: 'Front End Developer',
    linkSite: 'https://www.bluealba.com/',
    content: [
      {key: 1, text: 'Improved the interaction of brokers with all the data generated in the app, special focus on the collection of commissions and comparative charts.'},
    ],
    techStack: [
      {key: 1, label: 'JavaScript'},
      {key: 2, label: 'React.js'},
      {key: 3, label: 'Sass'},
      {key: 4, label: 'Yarn'},
      {key: 5, label: 'Git & Github'},
      {key: 6, label: 'Docker for local setup'}
    ],
  },
  {
    id: 6,
    title: 'Toolbox - The Unified TV Ecosystem',
    years: 'September 2016 - January 2018',
    occupation: 'Front End Developer',
    linkSite: 'https://www.toolboxtve.com/',
    content: [
      {key: 1, text: 'Contributed to building a digital content manager app (Like Netflix) called Experience.'},
      {key: 2, text: 'Developed the principal UI features such as the grid of movies and series, adaptation of the main player in multiple devices.'},
      {key: 3, text: 'Implemented the customized version for the main clients such as DirecTv Play for Latin America, Dist OTT (Mexico).'},
    ],
    techStack: [
      {key: 1, label: 'JavaScript'},
      {key: 2, label: 'React.js'},
      {key: 4, label: 'Yarn'},
      {key: 5, label: 'Bootstrap'},
      {key: 6, label: 'Sass'},
      {key: 7, label: 'Git & Github'},
      {key: 8, label: 'UI/UX analysis'},
    ],
  },
  {
    id: 7,
    title: 'Crandi',
    years: 'January 2016 - September 2016',
    occupation: 'Front End Developer',
    linkSite: 'https://www.crandi.com/',
    content: [
      {key: 1, text: 'Participated in the team in the development team at this agency.'},
      {key: 2, text: 'Developed websites, landings pages, email marketing, and any web implementation for the digital campaigns.'},
    ],
    techStack: [
      {key: 1, label: 'HTML5'},
      {key: 2, label: 'CSS3'},
      {key: 3, label: 'Sass'},
      {key: 4, label: 'JavaScript'},
      {key: 5, label: 'Git'},
      {key: 7, label: 'Wordpress'},
      {key: 8, label: 'Responsive Design'},
    ],
  },
  {
    id: 8,
    title: 'Nextperience',
    years: 'December 2014 - November 2015',
    occupation: 'Front End Developer',
    linkSite: 'https://nextperience.net/',
    content: [
      {key: 1, text: 'Participated in the team in the performance team at this agency.'},
      {key: 2, text: 'Developed websites, landings pages, email marketing, and any web implementation for the digital campaigns.'},
    ],
    techStack: [
      {key: 1, label: 'HTML5'},
      {key: 2, label: 'CSS3'},
      {key: 3, label: 'JavaScript'},
      {key: 4, label: 'Git'},
      {key: 5, label: 'Wordpress'},
      {key: 6, label: 'Responsive Design'},
    ],
  },
]