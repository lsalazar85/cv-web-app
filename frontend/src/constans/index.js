const MY_PHOTO = 'https://scontent.faep33-1.fna.fbcdn.net/v/t1.6435-9/170810186_10157729943471816_8716995957618713047_n.jpg?_nc_cat=105&ccb=1-3&_nc_sid=8bfeb9&_nc_ohc=bk3yltZXhawAX_TZpwc&_nc_oc=AQnTLSgyp9e2SaB_qtrOUtHJmOauf1den9aC6EiDHvI9eSiPMlNYtTD96zUMAkLpfFN_zohwTTcA5YJq4PmHqOO8&_nc_ht=scontent.faep33-1.fna&oh=84d48ceeb56e34e2c7fb8e39c99d63e1&oe=60A35B5D';

export { MY_PHOTO }