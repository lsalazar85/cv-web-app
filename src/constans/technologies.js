const Technologies = [
  {
    javaScript: 'JavaScript',
    reactJs: 'React.js',
    reactNative: 'React Native',
    context: 'React Context State',
    redux: 'Redux',
    nodeJs: 'Node.js',
    express: 'Express',
    graphQl: 'GraphQl',
    mongoDb: 'Mongo DB',
    less: 'LESS',
    sass: 'SASS',
    styledComponents: 'Styled Components',
    html: 'HTML5',
    css: 'CSS3',
    gitHub: 'GitHub',
  }
]
export default (Technologies)