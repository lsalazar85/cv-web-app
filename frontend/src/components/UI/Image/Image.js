import React from 'react'
import { string, bool } from 'prop-types'

import { ImageContainer } from './styled'

const Image = ({ width, alt, src, circular }) => (
  <ImageContainer width={width} circular={circular}>
    <img src={src} alt={alt} />
  </ImageContainer>
)

Image.defaultProps = {
  circular: false,
}

Image.propTypes = {
  width: string.isRequired,
  src: string.isRequired,
  alt: string.isRequired,
  circular: bool,
}

export default Image
