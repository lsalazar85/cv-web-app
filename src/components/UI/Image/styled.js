import styled from 'styled-components'

const ImageContainer = styled.div`
    width: ${props => props.width};
    height: auto;
    
    img {
        width: 100%;
        height: auto;
        border-radius:${props => props.circular && '50%'};
    }
`

export { ImageContainer }