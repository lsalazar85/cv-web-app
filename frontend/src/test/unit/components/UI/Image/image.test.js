import React from 'react'
import { render, cleanup } from '@testing-library/react'

import Image from '../../../../../components/UI/Image/Image'
import { MY_PHOTO } from '../../../../../constans'

describe('Image Component', () => {
  afterEach(cleanup)

  it('should render correctly', () => {
    const { asFragment } = render(
      <Image src={MY_PHOTO} alt={'Avatar Image'} width={'75%'} />
    )
    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly with circular prop', () => {
    const { asFragment } = render(
      <Image src={MY_PHOTO} alt={'Avatar Image'} width={'75%'} circular />
    )
    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
})
