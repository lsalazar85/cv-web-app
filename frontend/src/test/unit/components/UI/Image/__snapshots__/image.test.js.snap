// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx kyfPvg"
    width="75%"
  >
    <img
      alt="Avatar Image"
      src="https://scontent.faep4-1.fna.fbcdn.net/v/t1.0-9/118404240_10157203921491816_5452134641605056529_n.jpg?_nc_cat=102&ccb=2&_nc_sid=09cbfe&_nc_ohc=YOJEOCJ3ZqUAX91suA2&_nc_ht=scontent.faep4-1.fna&oh=5a2c7ab5853c092d1964dbb100c43664&oe=6043CDCD"
    />
  </div>
</DocumentFragment>
`;

exports[`Image Component should render correctly with circular prop 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx MxgGA"
    width="75%"
  >
    <img
      alt="Avatar Image"
      src="https://scontent.faep4-1.fna.fbcdn.net/v/t1.0-9/118404240_10157203921491816_5452134641605056529_n.jpg?_nc_cat=102&ccb=2&_nc_sid=09cbfe&_nc_ohc=YOJEOCJ3ZqUAX91suA2&_nc_ht=scontent.faep4-1.fna&oh=5a2c7ab5853c092d1964dbb100c43664&oe=6043CDCD"
    />
  </div>
</DocumentFragment>
`;
